views/part-upsell.php

```diff
@@ -0,0 +1,85 @@
+<?php
+
+use Imagify\User\User;
+
+defined( 'ABSPATH' ) || die( 'Cheatin’ uh?' );
+
+$imagify_user = new User();
+$unconsumed_quota = $imagify_user ? $imagify_user->get_percent_unconsumed_quota() : 0;
+$infinite     = ( 15 === $imagify_user->plan_id || 17 === $imagify_user->plan_id );
+$upgrade      = '';
+$price        = '';
+$upgrade_link = '';
+$user_id      = get_current_user_id();
+$notices      = get_user_meta( $user_id, '_imagify_ignore_notices', true );
+$notices      = $notices && is_array( $notices ) ? array_flip( $notices ) : [];
+
+if (
+	$unconsumed_quota > 20
+	&&
+	isset( $notices['upsell-banner'] )
+) {
+	unset( $notices['upsell-banner'] );
+	$notices = array_flip( $notices );
+	$notices = array_filter( $notices );
+	$notices = array_values( $notices );
+
+	update_user_meta( $user_id, '_imagify_ignore_notices', $notices );
+}
+
+if (
+	Imagify_Requirements::is_api_key_valid()
+	&&
+	! $infinite
+	&&
+	! isset( $notices['upsell-banner'] )
+	&&
+	$unconsumed_quota <= 20
+	) {
+	?>
+<div class="imagify-col-content imagify-upsell">
+	<div class="imagify-flex imagify-vcenter">
+		<span class="imagify-meteo-icon imagify-noshrink"><?php echo $this->get_quota_icon(); ?></span>
+		<div class="imagify-space-left imagify-full-width">
+			<p>
+				<?php
+				printf(
+					/* translators: %s is a data quota. */
+					__( 'You have %s space credit left', 'imagify' ),
+					'<span class="imagify-unconsumed-percent">' . $this->get_quota_percent() . '%</span>'
+				);
+				?>
+			</p>
+
+			<div class="<?php echo $this->get_quota_class(); ?>">
+				<div class="imagify-unconsumed-bar imagify-progress" style="width: <?php echo $this->get_quota_percent() . '%'; ?>;"></div>
+			</div>
+		</div>
+	</div>
+	<?php
+	if ( $imagify_user->is_free() ) {
+		$upgrade = esc_html__( 'Upgrade your plan now for more!', 'imagify' );
+		$price = esc_html__( 'From $4.99/month only, keep going with image optimization!', 'imagify' );
+		$upgrade_link = 'https://app.imagify.io/subscription/?utm_source=plugin&utm_medium=upsell_banner';
```

Comment:
We have a constant called `IMAGIFY_APP_DOMAIN` that we can use instead of hard-coding the url, I think this will help with testing with staging for example, what do u think?